lib/models/LifecycleConfiguration.ts

```diff
@@ -673,6 +675,13 @@ export default class LifecycleConfiguration {
             const msg = 'Date must be in ISO 8601 format';
             return errors.InvalidArgument.customizeDescription(msg);
         }
+        const midnightRegex = new RegExp('^(-?(?:[1-9][0-9]*)?[0-9]{4})-' +
+            '(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T00' +
+            ':00:00(.[0-9]+)?(Z)?$');
+        if (!midnightRegex.test(date)) {
+            const msg = 'Date must have time set to midnight';
```

Comment:
AWS response: `<Error><Code>InvalidArgument</Code><Message>'Date' must be at midnight GMT</Message><ArgumentName>Date</ArgumentName><ArgumentValue>Wed Dec 31 16:00:01 PST 1969</ArgumentValue><RequestId>3FJ1W5R1WTMT5XSD</RequestId><HostId>b4XDlJx/XHjphwLBfZUbqFRXZEzIzkO1+bYYKsHw7a7S21OzODiPq3D/TjduHTWCqRfGiPplSS4=</HostId></Error>`

Today in S3C, I don't think we support ArgumentName/ArgumentValue but we should try to match the AWS error message: "'Date' must be at midnight GMT"